### ISSUE TITLE
l1 regularization

### ISSUE THREAD

**Issue by forcoogle** [eriklindernoren/ML-From-Scratch#58]
Shouldn't `class l1_regularization()` be defined as `self.alpha * np.linalg.norm(w, 1)` instead of `self.alpha * np.linalg.norm(w)`?

**Comment by cpy18727** [comment]
i think so too

**Comment by klintan** [comment]
Yes, I was looking at that for way too long. It should def be `np.linalg.norm(w, 1)` for the L1-norm or Manhattan distance, the default is the L2-norm (Frobenius).
